### PR TITLE
Implement suggestions

### PR DESCRIPTION
**Pull request by myk-lop**
Implement suggestions from review https://github.com/myk-lop/gridu-react/pull/1/files

### CODE REVIEW

File: twitter-clone/src/redux/reducers/tweetsSlice.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ const tweetsSlice = createSlice({
   name: "tweets",
   initialState,
   reducers: {
-    setTweets(state, action) {
+    setTweets: (state, action: PayloadAction<ITweet[]>) => {
```
Comment (yanisneverlies) [inline twitter-clone/src/redux/reducers/tweetsSlice.ts]: we also need a type for state in these functions

File: twitter-clone/src/pages/home/components/TweetsList/TweetsList.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import styles from "./TweetsList.module.scss";
+import { RootState } from "../../../../redux/configureStore";
+
+const TweetsList = () => {
+  const tweets = useSelector((state: RootState) => state.tweets);
+
+  return (
+    <ul className={styles.list}>
+      {tweets.map(({ id, text, author_id }: ITweet) => (
+        <Tweet key={id} author_id={author_id} text={text} />
```
Comment (yanisneverlies) [inline twitter-clone/src/pages/home/components/TweetsList/TweetsList.tsx]: we usually name props using camelCase, so here we would have `authorId`

File: twitter-clone/src/common/components/Header/Header.tsx [hunk 3]
```diff
@@ -21,7 +14,7 @@ const Header = ({ userName }: { userName: string }) => {
   return (
     <header className={styles.header}>
       <div className={styles.userName}>{userName}</div>
-      <Button onClick={logOut} additionalStyles={buttonStyles}>
+      <Button onClick={logOut} styleType="secondary">
```
Comment (yanisneverlies) [inline twitter-clone/src/common/components/Header/Header.tsx]: style type can be renamed as 'variant'. this approach is used in Chakra UI for example, can be more informative

**Review by yanisneverlies — COMMENTED** [review]
The code looks cleaner and more readable, great job!

though I left a few comments, they are minor and would be easy to implement

**Comment by yanisneverlies** [comment]
one very important thing that was not implemented in this PR: you're using `utils` folder to store react components, which is not correct. usually, we're using `utils` to store functions/variables that we use in the components. So you also need to move `RequireAuth` component from `utils` folder and move your util function back to this folder

**Comment by myk-lop** [comment]
Implemented all suggestions, merging branch to main

**Pull request merged**